### PR TITLE
Add support for partial schemas in service:push

### PR DESCRIPTION
**Pull request by JakeDawkins**
TODO:

- [ ] Update CHANGELOG.md\* with your change (include reference to issue & this PR)
- [ ] Make sure all of the significant new logic is covered by tests
- [ ] Rebase your changes on master so that they can be merged easily
- [ ] Make sure all tests and linter rules pass

\*Make sure changelog entries note which project(s) has been affected. See older entries for examples on what this looks like.

### CODE REVIEW

File: packages/apollo-language-server/src/providers/federation-info/endpoint.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      );
+    }
+
+    this.info = data._service;
+  }
+
+  async resolveFederationInfo() {
+    if (!this.info) await this.getFederationInfo();
+
+    if (!this.info) throw new Error("No service info available");
```
Comment (trevor-scheer) [inline packages/apollo-language-server/src/providers/federation-info/endpoint.ts]: I understand what's going on here, but there's a level of indirection around `this.info` that I think we could remove. If `getFederationInfo` just returned `data._service`, we could handle the `this.info` assignment here instead, and it'll be more obvious.

```js
if (!this.info) {
  this.info = ...
}
```

File: packages/apollo/src/commands/service/push.ts [hunk 2]
```diff
@@ -28,6 +35,13 @@ export default class ServicePush extends ProjectCommand {
       {
         title: "Uploading service to Engine",
         task: async () => {
+          // handle partial schema uploading
+          if (flags.federated) {
+            const info = await (project as GraphQLServiceProject).resolveFederationInfo();
```
Comment (trevor-scheer) [inline packages/apollo/src/commands/service/push.ts]: I think if we leverage the generic types for `this.runTasks` we _might_ be able to get rid of the cast here. See `ClientCheck` or `ServiceCheck` for an example.

File: packages/apollo-language-server/src/providers/federation-info/index.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    // }
+    if (config.service.endpoint) {
+      return new EndpointFederationInfoProvider(config.service.endpoint);
+    }
+    throw new Error(
+      "You must provide a service endpoint to use the federation info provider"
+    );
+  }
+
+  if (isClientConfig(config)) {
```
Comment (trevor-scheer) [inline packages/apollo-language-server/src/providers/federation-info/index.ts]: This is great 💯

File: packages/apollo-language-server/src/providers/federation-info/endpoint.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  private async getFederationInfo() {
+    const { skipSSLValidation, url, headers } = this.config;
+    const options: HttpLink.Options = {
+      uri: url,
+      fetch,
+      ...(skipSSLValidation && { fetchOptions: { agent: new Agent() } })
+    };
+
+    const getFederationInfoQuery = `
+      query getFederationInfo {
```
Comment (trevor-scheer) [inline packages/apollo-language-server/src/providers/federation-info/endpoint.ts]: Iiiiiinteresting...just thinking out loud. Seems we have a meta graphql problem here. Our type generation and VSCode are both going to try to validate this query against our schema 😂 

Might need an `// apollo-ignore` at some point.

Comment (JakeDawkins) [reply packages/apollo-language-server/src/providers/federation-info/endpoint.ts]: Not in this case :) There's no `gql` so it's happily ignored!

**Pull request closed without merging**